### PR TITLE
Add support for revoking authorizations.

### PR DESCRIPTION
**Pull request by erikwiffin**

### CODE REVIEW

File: auth_proxy/models/user.py [hunk 1]
```diff
@@ -16,3 +17,26 @@ class User(Base):
...
+        """
+        return str(self.id)
+
+    def is_authenticated(self):
+        """ Return True if the user is authenticated.
+        """
+        return self.authenticated
+
+    def is_anonymous(self):
+        """ False, as anonymouse users aren't supported.
```
Comment (jmandel) [inline auth_proxy/models/user.py]: **anonymous** :p

File: auth_proxy/models/user.py [hunk 2]
```diff
@@ -16,3 +17,26 @@ class User(Base):
...
+        """ True, as all users are active.
+        """
+        return True
+
+    def get_id(self):
+        """ Return the user id.
+        """
+        return str(self.id)
+
+    def is_authenticated(self):
```
Comment (jmandel) [inline auth_proxy/models/user.py]: I can't figure out where this property is set or the accessor accessed. Is there framework magic here?

File: auth_proxy/models/user.py [hunk 3]
```diff
@@ -16,3 +17,26 @@ class User(Base):
 
     id = Column(Integer, primary_key=True)
     patient_id = Column(String)
+    username = Column(String)
+    password = Column(String)
```
Comment (jmandel) [inline auth_proxy/models/user.py]: This is a little silly, but I'm still weirded out seeing even a fixturized password stored plain in a db. Would you object to hashing (e.g. with https://pypi.python.org/pypi/bcrypt/3.1.0)?

File: auth_proxy/static/main.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        });
+
+    $.each($el.data(), function (key, value) {
+      $form.append($('<input />', {
+        name: key,
+        value: value
+      }));
+    });
+
+    if (confirm('Are you sure?')) {
```
Comment (jmandel) [inline auth_proxy/static/main.js]: This confirm should be replaced with a dialog like https://docs.google.com/presentation/d/1kQU0_dDL2vNdAuI4iUatthSTo7BxV_Hrtx2LuO4D6p8/edit#slide=id.p43

image

Happy to merge first and do that later if you prefer; let me know.

File: auth_proxy/models/user.py (hunk 2 above)
Comment (erikwiffin) [inline auth_proxy/models/user.py]: Yeah, it's required by https://flask-login.readthedocs.io/en/latest/#your-user-class. I doesn't need to be stored in the db though, so I'll fix that.

File: auth_proxy/static/main.js (hunk 4 above)
Comment (erikwiffin) [inline auth_proxy/static/main.js]: There's a typo in your slide, "form Healthcare Provider".

File: auth_proxy/models/user.py (hunk 3 above)
Comment (erikwiffin) [inline auth_proxy/models/user.py]: Using http://sqlalchemy-utils.readthedocs.io/en/latest/data_types.html#module-sqlalchemy_utils.types.password.

File: auth_proxy/static/main.js (hunk 4 above)
Comment (jmandel) [inline auth_proxy/static/main.js]: fixed, thanks!

File: auth_proxy/models/user.py (hunk 3 above)
Comment (jmandel) [inline auth_proxy/models/user.py]: I'm seeing a `String` column rather than `PasswordType` -- probably I'm just missing something.

Comment (jmandel) [inline auth_proxy/models/user.py]: Oh, I see -- I'm looking at the first commit here, and you fixed it in the 2nd.

**Pull request merged**